src/lib/SourceRename.hs

```diff
@@ -86,34 +99,44 @@ class SourceRenamableB (b :: B) where
 
 instance SourceRenamableE (SourceNameOr UVar) where
   sourceRenameE (SourceName sourceName) =
-    InternalName <$> lookupSourceName sourceName
+    InternalName sourceName <$> lookupSourceName sourceName
   sourceRenameE _ = error "Shouldn't be source-renaming internal names"
 
 lookupSourceName :: Renamer m => SourceName -> m n (UVar n)
 lookupSourceName v = do
-  SourceMap sourceMap <- askSourceMap
-  case M.lookup v sourceMap of
-    Nothing -> throw UnboundVarErr $ pprint v
-    Just v' -> return v'
+  local <- askLocalSourceMap
+  case lookupSourceMapPure local v of
+    [] -> do
+      imported <- askImportedSourceMap
+      case lookupSourceMapPure imported v of
+        [] -> throw UnboundVarErr $ pprint v
+        [SourceNameDef maybeV _] -> case maybeV of
+          Just v' -> return v'
+          Nothing -> throw VarDefErr v
+        _ -> throw AmbiguousVarErr $ pprint v
+    [SourceNameDef maybeV _] -> case maybeV of
+      Just v' -> return v'
+      Nothing -> throw VarDefErr v
+    _ -> throw AmbiguousVarErr $ pprint v
```

Comment:
This is a little surprising. Local vars are allowed to unambiguously shadow imported vars? That's not what I would expect and I don't think many languages have such scoping rules

Reply:
Fair point. I'll make that case an error.